# Statistics 
Statistic is a number that summarizes the data collected from sample. 
There are number of ways to do that:
1. average
2. percentage 
3. percentile etc 

### Rate :
A rate is a ratio that reflects some quantity per certain unit.

The lower the response rate, the less credible the results and more the results will favour the opinion of those who responded.(and respondents tend to have stonger opinions about the issue they choose to respond to.)

### Population :
The group of individual that you wish to study in order to answer your research question

### Sample :
If you select small number of individuals from the population to study those individuals and use that information to draw conclusions about the whole population, then it will be called as sample. 

### Bias :
Bias is favourism that is present in the data collection process, resulting in lopsided, misleading results. 
Bias can occure in any number of ways.
e.g.
1. In the way the sample is selected 
2. In the way the data is collected (Specially type of questions that were asked)


## Types of data :


#### 1.  Numerical Data / Quantitative Data / Measurement Data :
It is data that have meaning as a measurement. e.g. person's height, weight,IQ, number of stocks that person owns etc.

#### 2. Categorical data / Qualitative Data :
repesents characteristics such as person's gender, opinion, race etc.


### Dataset :
A dataset is the collection of all data taken from your sample. 

In [1]:
import pandas as pd

data = pd.read_csv("https://raw.githubusercontent.com/hannansatopay/DSML/master/Admission_Predict.csv")
data.head()

Serial No.  GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  \
0           1        337          118                  4  4.5   4.5  9.65   
1           2        324          107                  4  4.0   4.5  8.87   
2           3        316          104                  3  3.0   3.5  8.00   
3           4        322          110                  3  3.5   2.5  8.67   
4           5        314          103                  2  2.0   3.0  8.21   

   Research  Chance of Admit   
0         1              0.92  
1         1              0.76  
2         1              0.72  
3         1              0.80  
4         0              0.65

In [2]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 9 columns):
Serial No.           400 non-null int64
GRE Score            400 non-null int64
TOEFL Score          400 non-null int64
University Rating    400 non-null int64
SOP                  400 non-null float64
LOR                  400 non-null float64
CGPA                 400 non-null float64
Research             400 non-null int64
Chance of Admit      400 non-null float64
dtypes: float64(4), int64(5)
memory usage: 28.2 KB


### Mean(Average) :
The mean, is used to measure the center or middle of a numerical dataset. It is sum of all numbers divided by total numbers.

 $\bar {x}={\frac {1}{n}}\left(\sum _{i=1}^{n}{x_{i}}\right)={\frac {x_{1}+x_{2}+\cdots +x_{n}}{n}}$
 
 The average of dataset is affected by outliers but the median is not. 
 
 When calculating mean of *sample dataset* , x_i should be divided by *n* and not n-1.

#### A. Arithmatic mean 
1. It is average of group of numbers.
2. Applicable for interval and ratio data.
3. Affected by outliers.

In [3]:
# Find out mean\ Average GRE score and TOEFL score 
gre=data["GRE Score"]
toefl=data["TOEFL Score"]

In [4]:
# Average GRE score
GRE_average=gre.mean()
GRE_average

316.8075

In [5]:
# Average TOEFL score 
TOEFL_average=toefl.mean()
TOEFL_average

107.41

In [6]:
# Find out students whose score is above gre average 
gre_above_avg = gre >= GRE_average
gre_above_avg.value_counts()


True     205
False    195
Name: GRE Score, dtype: int64

In [7]:
# Find out students whose score is above toefl average 
toefl_above_avg= toefl >= TOEFL_average 
toefl_above_avg.value_counts()

False    209
True     191
Name: TOEFL Score, dtype: int64

In [8]:
data.mean(axis=0)

Serial No.           200.500000
GRE Score            316.807500
TOEFL Score          107.410000
University Rating      3.087500
SOP                    3.400000
LOR                    3.452500
CGPA                   8.598925
Research               0.547500
Chance of Admit        0.724350
dtype: float64

### Median :

Median of the dataset is the value that lies exactly in the middle. 
In a numerical dataset, the median is the point at which there are an equal number of datapoints whose values lie above and below the median value. Thus median is truly median of dataset. 
e.g. median of interstate highway.On a highway, median is the middle of the road and an equal number of lanes lay on either side of the median. 

Steps for finding the median of the dataset :
1. Order the number from smallest to largest 
2. If the dataset contains odd numbers, choose the one exactly in the middle 
   Otherwise ,
   take 2 numbers that appear exactly in the middle and average them to find the median .
   

In [9]:
# GRE median 
GRE_median = gre.median()
GRE_median

317.0

In [10]:
# TOEFL median 
TOEFL_median = toefl.median()
TOEFL_median

107.0

In [11]:
data.median()

Serial No.           200.50
GRE Score            317.00
TOEFL Score          107.00
University Rating      3.00
SOP                    3.50
LOR                    3.50
CGPA                   8.61
Research               1.00
Chance of Admit        0.73
dtype: float64

The average and median are two different representation of same data and can ofter give two very different stories about data. 

### Mode :
Mode is the most frequent value that appear in the dataset 

### Standard deviation :

It is a way to measure variability(or spread) among the numbers in a data set. As a term implies, it is a standard amount of deviation(or distance) from average. So it is average distance from mean. 
It describes where the most data should fall.


The formula for sample standard deviation  :    
                         
${\displaystyle s={\sqrt {{\frac {1}{n-1}}\sum _{i=1}^{n}\left(x_{i}-{\bar {x}}\right)^{2}}}.}$ 


Where ,                                                                      
$s	=$	sample standard deviation                                                                                  
$n	=$	the size of the sample                                                                             
$x_i	=$	each value from the population sample                                                                                
$\bar {x}	=$	average of all values                           

- The standard deviation of entire population is denoted by greek letter v. 
- The standard deviation of a sample population is denoted with the letter s. 
Because most of the time the population standard deviation isn't a value that's known. 

To calculate the sample standard deviation,s, do the following steps :
1. Find the average of dataset
2. Take each number and subtract average from it.
3. Square each of the differences.
4. Add up all the results of step 3. 
5. Divide the squares by total numbers in the dataset minus one(n-1).
6. Take the square root of the number you get.

Why n-1 ?
when it comes to sample standard deviation, dividing by n-1 makes sure that the standard deviation isn't biased on average. 


A small standard deviation means that the values in the dataset are closed to the average of dataset. 

#### Uses of standard deviation :
1. Indicator od finacial risk
2. Comparing population :                            
   a. household income in two cities                                   
   b. employee absenteeism at two plants                                                    


In [12]:
gre.std()

11.473646113500477

In [13]:
gre.describe()

count    400.000000
mean     316.807500
std       11.473646
min      290.000000
25%      308.000000
50%      317.000000
75%      325.000000
max      340.000000
Name: GRE Score, dtype: float64

### Percentile :
The percentile reported for given score is the percentage of values in the dataset that fall below that certain score.
Generally percentiles are used for comparison purposes to determine relative standing. 
It measures central tendency that divides a group of data into 100 parts.
e.g. 90th percentile indicates that most 90% data lie below it and only 10% data lie above it.

#### How to calculate percentile ?
1. Organize datainto ascending array
2. Calculate Pth percentile location :                              
      i = P/100*n                           
      where,                              
      i = location index                            
      n = total values in column 

In [14]:
# sort gre by descending order
gre_descending = gre.sort_values(ascending=False)
gre_descending.head()

284    340
143    340
384    340
202    340
25     340
Name: GRE Score, dtype: int64

In [15]:
gre_descending.tail()

79     294
272    294
168    293
377    290
117    290
Name: GRE Score, dtype: int64

In [16]:
# Sort toefl by descending order 
toefl_descending = toefl.sort_values(ascending=False) 
toefl_descending.head()

297    120
81     120
143    120
202    120
203    120
Name: TOEFL Score, dtype: int64

In [17]:
toefl_descending.tail()

272    95
347    94
28     93
79     93
368    92
Name: TOEFL Score, dtype: int64

So, as we can see, highest gre score is 340 and highest toefl score is 120 . Whereas lowest gre score is 290 and lowest toefl score is 92 .

### When to use mean, median, mode ?
When data is right\left skewed then use median as a central tendency , if its bell shaped curve then anything can be used

### Normal Distribution :
#### Empirical Rule ..... if the histogram is bell shaped
 1. Approx. 68% of all observation fall within one standard deviation of the mean.
 2. Approx. 95% of all abservations fall within two standard deviation of the mean.
 3. Approx. 99.7% of all observation fall within three standard deviation of the mean.